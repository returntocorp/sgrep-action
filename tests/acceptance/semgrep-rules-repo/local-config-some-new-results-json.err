| versions          - semgrep x.y.z on Python x.y.z
| environment       - running in environment git, triggering event is 'unknown'
| manage            - not logged in
=== setting up agent configuration
| using semgrep rules from python/django/security/audit/xss/template-translate-as-no-escape.yaml
| using default path ignore rules of common test and dependency directories
| scanning only the following commits:
|   * XSS via translate filters in Django Templates (#1071)
| looking at 6 changed paths
| found 6 files in the paths to be scanned
=== looking for current issues in 6 files
| 2 current issues found
| No ignored issues found
=== not looking at pre-existing issues since all files with current issues are newly created
=== exiting with failing status
