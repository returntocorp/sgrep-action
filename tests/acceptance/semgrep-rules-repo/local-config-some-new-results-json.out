[{"check_id": "python.django.security.audit.xss.template-translate-as-no-escape", "path": "python/django/security/audit/xss/template-translate-as-no-escape.html", "line": 8, "column": 1, "message": "Translated strings will not be escaped when rendered in a template.\nThis leads to a vulnerability where translators could include malicious script tags in their translations.\nConsider using `force_escape` to explicitly escape a transalted text.\n", "severity": 0, "index": 0, "end_line": 15, "end_column": 15, "commit_date": "2021-01-19T16:19:23+10:00", "metadata": {"cwe": "CWE-79: Improper Neutralization of Input During Web Page Generation ('Cross-site Scripting')", "owasp": "A7: Cross-site Scripting (XSS)", "references": ["https://edx.readthedocs.io/projects/edx-developer-guide/en/latest/preventing_xss/preventing_xss_in_django_templates.html#html-escaping-translations-in-django-templates", "https://docs.djangoproject.com/en/3.1/topics/i18n/translation/#internationalization-in-template-code"]}, "syntactic_id": "1c1e03f68c2c669feaedef412935de8e", "is_blocking": true}, {"check_id": "python.django.security.audit.xss.template-translate-as-no-escape", "path": "python/django/security/audit/xss/template-translate-as-no-escape.html", "line": 2, "column": 1, "message": "Translated strings will not be escaped when rendered in a template.\nThis leads to a vulnerability where translators could include malicious script tags in their translations.\nConsider using `force_escape` to explicitly escape a transalted text.\n", "severity": 0, "index": 0, "end_line": 20, "end_column": 20, "commit_date": "2021-01-19T16:19:23+10:00", "metadata": {"cwe": "CWE-79: Improper Neutralization of Input During Web Page Generation ('Cross-site Scripting')", "owasp": "A7: Cross-site Scripting (XSS)", "references": ["https://edx.readthedocs.io/projects/edx-developer-guide/en/latest/preventing_xss/preventing_xss_in_django_templates.html#html-escaping-translations-in-django-templates", "https://docs.djangoproject.com/en/3.1/topics/i18n/translation/#internationalization-in-template-code"]}, "syntactic_id": "6fd3006b3f0af31deea70ac866ffe7c3", "is_blocking": true}]
